File: auto_test.c
#include <stdio.h>
#include <stdlib.h>
int main(int argc, char *argv[])
{
    int T = atoi(argv[1]);
    for (int i = 1; i < T; i++)
    {
        printf("The result of No. %d case is: ", i);
        system("./data.out > data.in");
        system("./totest.out < data.in > totest.txt");
        system("./std.out < data.in > std.txt");
        if (system("diff std.txt totest.txt"))
        {
            printf("\033[1;31mWrong Answer\033[0m\n");
            return 0;
        }
        else
            printf("\033[1;32mAccepted\033[0m\n");
    }
    return 0;
}

File: auto_test.py
import sys
import os
T = int(sys.argv[1])
for i in range(1, T + 1):
    print('The result of No.', i, 'case is: ', end = '')
    os.system('python3 data.py > data.in')
    os.system('python3 totest.py < data.in > totest.txt')
    os.system('python3 std.py < data.in > std.txt')
    if os.system('diff std.txt totest.txt'):
        print('\033[1;31mWrong Answer\033[0m')
        break
    else:
        print('\033[1;32mAccepted\033[0m')
